src/main.rs

```diff
@@ -123,11 +123,30 @@ impl State {
     /// Initialize the window and its EGL surface.
     fn init_window(&mut self, connection: &mut Connection, queue: &QueueHandle<Self>) {
         // Initialize EGL context.
-        let native_display = NativeDisplay::new(connection.display());
-        let display = EGLDisplay::new(&native_display, None).expect("Unable to create EGL display");
-        let context =
-            EGLContext::new_with_config(&display, GL_ATTRIBUTES, Default::default(), None)
-                .expect("Unable to create EGL context");
+        let mut raw_display = WaylandDisplayHandle::empty();
+        raw_display.display = connection.backend().display_ptr().cast();
+        let raw_display = RawDisplayHandle::Wayland(raw_display);
+
+        let display =
+            unsafe { Display::from_raw(raw_display).expect("Unable to create EGL display") };
+
+        let config_template = ConfigTemplateBuilder::new().with_api(Api::GLES2).build();
+        let config = unsafe {
+            display
+                .find_configs(config_template)
+                .ok()
+                .and_then(|mut configs| configs.next())
+                .expect("No suitable configuration found")
+        };
+
+        let context_attributes = ContextAttributesBuilder::new()
+            .with_context_api(ContextApi::Gles(Some(Version::new(2, 0))))
+            .build(None);
```

Comment:
Seems a bit strange to me that one needs to first create a template and specify the API, then specify the API again here? Especially because both are passed to the same function in the end.

Also the `Api-specific` is broken in the docs here: https://docs.rs/glutin/0.30.0-beta.1/glutin/context/struct.ContextAttributesBuilder.html#method.build

Reply:
It's actually not, since the API is a bitset and config can support multiple Apis, like OpenGL and Gles at the same time, however context can have only one and with explicit version, so that's why they differ.